internal/middleware/etag_middleware.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+)
+
+var InvalidETagErr = errors.New("empty etag provided")
+
+func ETagMiddleware(expires time.Duration, name, etag string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			if etag == "" {
+				panic(InvalidETagErr)
+			}
+			etagValue := fmt.Sprintf("\"hms-%s-%s\"", name, etag)
```

Comment:
Why do we want to have the `hms` prefix? HMS is internal name and only RH, so I'd rather not expose it (even in minor stuff user really don't see). Can we have `provisioning`/`prov` if we want prefix it?

Reply:
Sure, I was aiming for something short, it's not mandatory it's recommended to use a prefix to differentiate own content (we do have a name). Changed this to "pb".